Sources/BraveUI/SwiftUI/WebImageReader.swift

```diff
@@ -5,81 +5,82 @@
-      with: url, options: options, progress: nil,
-      completed: { [weak self] image, data, error, _, finished, _ in
-        guard let self = self else { return }
-        self.image = image
-        self.error = error
-        if finished {
-          self.isFinished = true
-        }
-      })
+extension EnvironmentValues {
+  public var webImageDownloader: WebImageDownloaderType? {
+    get { self[WebImageDownloaderKey.self] }
+    set { self[WebImageDownloaderKey.self] = newValue }
   }
+}
 
-  func cancel() {
-    operation?.cancel()
-    operation = nil
+extension SDWebImageManager: WebImageDownloaderType {
+  @MainActor public func downloadImage(url: URL?) async -> UIImage? {
+    return await withCheckedContinuation { continuation in
+      loadImage(with: url, progress: nil) { image, _, _, _, _, _ in
+        continuation.resume(returning: image)
+      }
+    }
   }
   
-  func load(base64Str: String, options: [SDImageCoderOption: Any] = [:]) {
-    guard base64Str.hasPrefix("data:image/") else { return }
-    guard let dataString = base64Str.separatedBy(",").last else { return }
-    
-    let data = Data(base64Encoded: dataString, options: .ignoreUnknownCharacters)
-    for coder in supportedCoders where coder.canDecode(from: data) {
-      image = coder.decodedImage(with: data, options: options)
-      break
+  public func imageFromData(data: Data) -> UIImage? {
+    SDImageCodersManager.shared.decodedImage(with: data)
+  }
+}
+
+extension BraveCore.WebImageDownloader: WebImageDownloaderType {
+  @MainActor public func downloadImage(url: URL?) async -> UIImage? {
+    if let url {
+      return await withCheckedContinuation { continuation in
+        downloadImage(url) { image, _, _ in
+          continuation.resume(returning: image)
+        }
+      }
     }
+    return nil
+  }
+  
+  public func imageFromData(data: Data) -> UIImage? {
+    WebImageDownloader.image(from: data)
   }
 }
 
 public struct WebImageReader<Content: View>: View {
-  @StateObject private var imageManager: WalletWebImageManager = .init()
   var url: URL?
-  var options: SDWebImageOptions
-  var coderOptions: [SDImageCoderOption: Any]
+  
+  @Environment(\.webImageDownloader) private var imageDownloader: WebImageDownloaderType?
+  
+  @State private var image: UIImage?
 
-  private var content: (_ image: UIImage?, _ isFinished: Bool) -> Content
+  private var content: (_ image: UIImage?) -> Content
 
   public init(
     url: URL?,
-    options: SDWebImageOptions = [],
-    coderOptions: [SDImageCoderOption: Any] = [:],
-    @ViewBuilder content: @escaping (_ image: UIImage?, _ isFinished: Bool) -> Content
+    @ViewBuilder content: @escaping (_ image: UIImage?) -> Content
   ) {
     self.content = content
     self.url = url
-    self.options = options
-    self.coderOptions = coderOptions
   }
 
   public var body: some View {
-    content(imageManager.image, imageManager.isFinished)
+    content(image)
       .onAppear {
```

Comment:
Make this `.task` instead so it is cancelled when the view leaves the hierarchy and add `.id(url)` to the view to ensure its reloaded properly when the URL changes